pkg/sigv4/sigv4.go

```diff
@@ -0,0 +1,244 @@
+package sigv4
+
+import (
+	"bytes"
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+)
+
+type Signer interface {
```

Comment:
If I'm not mistaken, we could make this interface (and some other structs in this package) unexported, which explicitly marks them for internal only (package) usage, helping with possible refactoring in the future and protecting other usage